src/sgame/sg_admin.cpp

```diff
@@ -4194,35 +4194,45 @@ static int ban_out( void *ban, char *str )
 
 	date[ i ] = 0;
 
-	if ( !b->expires || b->expires - t > 0 )
+	if ( !b->expires )
 	{
 		G_admin_duration( b->expires ? b->expires - t : -1,
-						  time, sizeof( time ),
-						  duration, sizeof( duration ) );
+		                  time, sizeof( time ),
+		                  duration, sizeof( duration ) );
+		d_color = Color::Red;
+	}
+	else if ( b->expires - t > 0 )
+	{
+		G_admin_duration( b->expires ? b->expires - t : -1,
+		                  time, sizeof( time ),
+		                  duration, sizeof( duration ) );
 	}
 	else
 	{
 		*time = 0;
-		Q_strncpyz( duration, "expired", sizeof( duration ) );
-		d_color = Color::Cyan;
-	}
-
-	Com_sprintf( str, MAX_STRING_CHARS, "%s"
-	             "         %s\\__ %s%s%-*s %s%-15s ^7%-8s %s"
-	             "          %s\\__ %s:^* %s",
-	             b->name,
-	             Color::ToString( G_ADMIN_BAN_IS_WARNING( b ) ? Color::Yellow : Color::Red ).c_str(),
-	             Color::ToString( d_color ).c_str(),
-	             time,
-	             MAX_DURATION_LENGTH - 1,
-	             duration,
-	             Color::ToString( ( strchr( b->ip.str, '/' ) ) ? Color::Red : Color::White ).c_str(),
+		Q_strncpyz( duration, "(expired)", sizeof( duration ) );
+		d_color = Color::Green;
+	}
+
+	Com_sprintf( str, MAX_STRING_CHARS, 
+	             "^3------------------------------\n"
+	             "    ^7Name:     %s^7\n"
+	             "    ^7IP:       %s%s^7\n"
+	             "    ^7Admin:    %s^7\n"
+	             "    ^7Reason:   ^3%s^7\n"
+	             "    ^7Date:     ^3%s^7\n"
+	             "    ^7Details:  %s%s%s%s%s%s^7",
+	             b->name, 
+	             Color::ToString( ( strchr( b->ip.str, '/' ) ) ? Color::Red : Color::Yellow ).c_str(),
 	             b->ip.str,
-	             date,
 	             b->banner,
+	             b->reason,
+	             date, 
 	             Color::ToString( G_ADMIN_BAN_IS_WARNING( b ) ? Color::Yellow : Color::Red ).c_str(),
-	             G_ADMIN_BAN_IS_WARNING( b ) ? "WARNING" : "BAN",
-	             b->reason );
+	             G_ADMIN_BAN_IS_WARNING( b ) ? "WARNING " : "BAN ",
+	             ( atoi( time ) > 0 ? "^7- expiring in " : "^7- " ),
```

Comment:
Seems simpler to construct one string above that would combine `time` `d_color` and `duration` instead of this ugly atoi hack.

Reply:
I've removed the atoi hack while keeping the original string.